skepticoin/scripts/send.py

```diff
@@ -74,6 +75,10 @@ def main():
 
             # it's late and I'm too lazy for the efficient & correct implementation.
             coinstate = thread.local_peer.chain_manager.coinstate
+
+            # TODO mypy complains coinstate could be None here
```

Comment:
I'm having trouble parsing this, which of the below 2 is it?

mypy complains, that's why you added the assert and now it no longer complains

OR

mypy complained, I added this assert to get it to shut up

Reply:
Not sure if I went into detail about this, so I'll do it here.

What happens is that a field (`coinstate` in this case) can be `None`, because it's declared as such in the `__init__()` function.
Therefore the field will remain `Optional[...]` for the rest of the program.
The solution would be to make the coinstate value an argument of the `__init__()` function to rule out this problematic case. I just don't want to do that in this PR, because it's out of scope.

IMO this just highlights the fact that we need to address this. It may look ugly, but in practice this assert statement has the same value as `assert True`: it will not fail and is therefore effectively redundant.

But if we don't do this then mypy will complain. And I really don't want to put `type: ignore` everywhere where `coinstate` is used, as that defeats the purpose of using type-checking.